### PR TITLE
Update EVM tool documentation

### LINKED ISSUE #1235: Add cli flag `--json-alloc` related to EVM Tool
EVM Tool adds a new `--json-alloc` flag that will output the post-execution state to the standard out for both running execution and state-test execution. The format of the file follows the ethereum-test "T8n" format here - https://ethereum-tests.readthedocs.io/en/latest/t8ntool-ref.html#allocation-files - this is also the genesis file format used by besu for the "alloc" section.

This flag is specified in the same places --json can be specified.

See https://github.com/hyperledger/besu/pull/4709

### LINKED ISSUE #1236: Document subcommand `code-validate` for EVM tool
EVM tool gains a sub-command `code-validate` that follows the practices described here - https://github.com/holiman/txparse/#eof-parser-eofparse

Code in standard input may have spaces added for clarity. Code may also be inputted via the command line as parameters.

See https://github.com/hyperledger/besu/pull/4845

### LINKED ISSUE #1247: Add `--fork` to the EVM tool
Update the EVM tool cli options to include the `--fork` option.

See: https://github.com/hyperledger/besu/pull/4913

### PR DESCRIPTION
**Pull request by alexandratran**
Update EVM tool documentation, including adding documentation for:

- `--json-alloc`
- `--fork`
- `code-validate`
- t8ntool support (see @shemnon's comment below)

fixes #1235 
fixes #1236 
fixes #1247

### CODE REVIEW

File: docs/public-networks/reference/evm-tool.md [hunk 1]
```diff
@@ -153,7 +155,25 @@ The amount is not reduced to cover intrinsic cost and gas fees.
...
-Provide an operation-by-operation trace of the command in JSON when set to true.
+When set to `true`, provides an operation-by-operation trace of the command in JSON.
+The default is `false`.
+
+### `json-alloc`
+
+=== "Syntax"
+
+    ```bash
+    --json-alloc=<boolean>
```
Comment (shemnon) [inline docs/public-networks/reference/evm-tool.md:166]: Just `--json-alloc`.

"When the option is set a json summary of the the post-execution world state and allocations will be printed"

File: docs/public-networks/reference/evm-tool.md [hunk 2]
```diff
@@ -169,8 +189,9 @@ Provide an operation-by-operation trace of the command in JSON when set to true.
     --nomemory=true
```
Comment (shemnon) [inline docs/public-networks/reference/evm-tool.md:189]: just `--nomemory`

"When this option is set..."

**Review by shemnon — COMMENTED** [review]
Looks good, but there are some changes coming with `t8n` support that will impact the `state-test` subcommand and main CLI.

First, can we remove all references to `=true` for options?  For the evm tool all boolean CLI flags will have that implicitly.  It also makes bad ui.  Having to use `--nomemory=false` to _enable_ memory in tracing is bad UX.  No negation is better than double negation. Better to treat boolean flags as standalone.

Here's the t8n impacts, all but one are only relevant when tracing.
- The `--json` flag gains an alias, `--trace`.  Both flags do the same thing.

- `--nomemory` becomes in essence 4 new flags, as seen by the --help menu.
   picoCLI documents them as `--[no]memory, --trace.[no]memory`
   `--nomemory` and `--trace.nomemory` turn off the memory in traces.  And 
   `--memory` and `--trace.memory` turn on the memory in traces.  
   The default is that memory traces are disabled, i.e. as though `--nomemory` is passed in.
   
- `--trace.[no]stack`   
   Enables and disables the operand stack in tracing for each op.
   Default is to show stack, i.e. as though `--trace.stack` is passed in.

- `--trace.[no]returndata`
   Enables and disables showing the return data in tracing for each op when present. 
   Default is to not show return data, i.e. as though `--trace.noreturndata` is passed in
   
-  `--[no]time` 
   Enables and disables timing data in the trace summary line.  
   Useful for testing and differential evaluations.
   This applies outside of tracing.
   
These flags were added to support the `t8n` spec for reference testing.  We may want to document the existance of the`t8n` command, but I wouldn't document any of the flags or how it works, other than to point users to the "official" spec here - https://ethereum-tests.readthedocs.io/en/latest/t8ntool.html

**Review by shemnon — APPROVED** [review]

File: docs/public-networks/reference/evm-tool.md [hunk 3]
```diff
@@ -135,42 +143,84 @@ Corresponds to the `data` field of the transaction and is returned by  the `CALL
...
+
+### `[no]time`
+
+=== "Syntax"
+
+    ```bash
+    --notime, --time
+    ```
+
+Setting `--notime` disables including time data in the summary output.
```
Comment (bgravenorst) [inline docs/public-networks/reference/evm-tool.md:220]: Perhaps we should rephrase this. Took me a few times to understand.

**Review by bgravenorst — APPROVED** [review]
Minor comment. LGTM

**Pull request merged**